reactivesocket-client/src/main/java/io/reactivesocket/client/ClientBuilder.java

```diff
@@ -104,26 +106,55 @@ private ClientBuilder(
         );
     }
 
-    public ReactiveSocket build() {
-        if (source == null) {
-            throw new IllegalStateException("Please configure the source!");
-        }
-        if (connector == null) {
-            throw new IllegalStateException("Please configure the connector!");
-        }
+    public Publisher<ReactiveSocket> build() {
+        return subscriber -> {
+            subscriber.onSubscribe(new Subscription() {
+                private ScheduledFuture<?> scheduledFuture = null;
+                private AtomicBoolean cancelled = new AtomicBoolean(false);
 
+                @Override
+                public void request(long n) {
+                    if (source == null) {
+                        subscriber.onError(new IllegalStateException("Please configure the source!"));
+                        return;
+                    }
+                    if (connector == null) {
+                        subscriber.onError(new IllegalStateException("Please configure the connector!"));
+                        return;
+                    }
 
-        ReactiveSocketConnector<T> filterConnector = connector;
-        if (requestTimeout > 0) {
-            filterConnector = filterConnector
-                .chain(socket -> new TimeoutSocket(socket, requestTimeout, requestTimeoutUnit, executor));
-        }
-        filterConnector = filterConnector.chain(DrainingSocket::new);
-
-        Publisher<? extends Collection<ReactiveSocketFactory<T>>> factories =
-            sourceToFactory(source, filterConnector);
+                    ReactiveSocketConnector<T> filterConnector = connector;
+                    if (requestTimeout > 0) {
+                        filterConnector = filterConnector
+                            .chain(socket -> new TimeoutSocket(socket, requestTimeout, requestTimeoutUnit, executor));
+                    }
+                    filterConnector = filterConnector.chain(DrainingSocket::new);
+
+                    Publisher<? extends Collection<ReactiveSocketFactory<T>>> factories =
+                        sourceToFactory(source, filterConnector);
+                    LoadBalancer<T> loadBalancer = new LoadBalancer<>(factories);
+
+                    scheduledFuture = executor.scheduleAtFixedRate(() -> {
+                        if (loadBalancer.availability() > 0 && !cancelled.get()) {
+                            subscriber.onNext(loadBalancer);
+                            subscriber.onComplete();
+                            if (scheduledFuture != null) {
+                                scheduledFuture.cancel(true);
+                            }
+                        }
+                    }, 1L, 50L, TimeUnit.MILLISECONDS);
```

Comment:
nit: If we have a outage and no servers are available, this will spin. Should we exponentially backoff?

I think it is "good enough" for now.